.github/workflows/config.yml

```diff
@@ -11,183 +11,310 @@ on:
+          restore-keys: |
+            ${{ runner.os }}-${{ env.ELIXIR_VERSION }}-${{ env.OTP_VERSION }}-${{ env.MIX_ENV }}-explorer-npm-
+
+      - name: Conditionally build Explorer NPM Cache
+        if: steps.explorer-npm-cache.outputs.cache-hit != 'true'
+        run: npm install
         working-directory: apps/explorer
-      - run: npm install
-        working-directory: apps/block_scout_web/assets
-      - run: npm rebuild node-sass
+
+      - name: Restore Blockscout Web NPM Cache
+        uses: actions/cache@v2
+        id: blockscoutweb-npm-cache
+        with:
+          path: apps/block_scout_web/assets/node_modules
+          key: ${{ runner.os }}-${{ env.ELIXIR_VERSION }}-${{ env.OTP_VERSION }}-${{ env.MIX_ENV }}-blockscoutweb-npm-${{ hashFiles('apps/block_scout_web/assets/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ env.ELIXIR_VERSION }}-${{ env.OTP_VERSION }}-${{ env.MIX_ENV }}-blockscoutweb-npm-
+
+      - name: Conditionally build Blockscout Web NPM Cache
+        if: steps.blockscoutweb-npm-cache.outputs.cache-hit != 'true'
+        run: npm install
         working-directory: apps/block_scout_web/assets
-      - run: mix compile
-      - run: make
-        working-directory: "deps/libsecp256k1"
-      - name: Build assets
-        run: node node_modules/webpack/bin/webpack.js --mode development
-        working-directory: "apps/block_scout_web/assets"
+
   credo:
+    name: Credo
     runs-on: ubuntu-18.04
+    needs: build-and-cache
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-elixir@v1
         with:
           otp-version: '23.3.4.1'
           elixir-version: '1.11.4'
-      - run: mix local.hex --force
-      - run: mix local.rebar --force
-      - run: mix deps.get
-      - run: mix compile
-      - run: make
-        working-directory: "deps/libsecp256k1"
+      
+      - name: Restore Mix Deps Cache
+        uses: actions/cache@v2
+        id: deps-cache
+        with:
+          path: | 
+            deps
+            _build
+          key: ${{ runner.os }}-${{ env.ELIXIR_VERSION }}-${{ env.OTP_VERSION }}-${{ env.MIX_ENV }}-deps-mixlockhash-${{ hashFiles('mix.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ env.ELIXIR_VERSION }}-${{ env.OTP_VERSION }}-${{ env.MIX_ENV }}-deps-"
+
       - run: mix credo
+
   check_formatted:
+    name: Code formatting checks
     runs-on: ubuntu-18.04
+    needs: build-and-cache
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-elixir@v1
         with:
           otp-version: '23.3.4.1'
           elixir-version: '1.11.4'
-      - run: mix local.hex --force
-      - run: mix local.rebar --force
-      - run: mix deps.get
-      - run: mix compile
-      - run: make
-        working-directory: "deps/libsecp256k1"
+
+      - name: Restore Mix Deps Cache
```

Comment:
Did you look into composite actions that could allow us extract these restore cache steps into a local action and avoid repetition like described [here](https://github.com/actions/runner/issues/646#issuecomment-901336347)?

Reply:
That seems pretty cool

Reply:
It looks like we can't use it right now due to this feature not supporting conditionals. We actually want to do the same as  https://github.com/actions/runner/issues/834#issuecomment-901847269 and create a wrapper around the cache action

Reply:
Ah, true, missed that.